mmv1/third_party/terraform/tpgresource/common_diff_suppress.go.erb

```diff
@@ -293,6 +293,19 @@ func ProjectNumberDiffSuppress(_, old, new string, _ *schema.ResourceData) bool
 	return a2 == b2
 }
 
+// Suppress diffs when the value read from api
+// has the project number instead of the project name.
+// This function suppresses the difference when the project ID is not prefixed with the string "projects".
+func ProjectNumberDiffSuppressWithoutPrefix(_, old, new string, _ *schema.ResourceData) bool {
```

Comment:
Please move this into a [`custom_code.constants` file](https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#add-reusable-variables-and-functions).

Also, please add [unit tests](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-unit-tests). And make sure that this handles the "list size" diff - that is, if the first value ends in `.#`, never suppress the diff.

Reply:
On discussion with my team, we have decided to support only project numbers for now and support project IDs later. I have made the relevant changes, could you please take a look, thank you!